src/main/java/com/academia/library/validator/OrderValidator.java

```diff
@@ -0,0 +1,65 @@
+package com.academia.library.validator;
+
+import com.academia.library.dto.OrderDetailsRequest;
+import com.academia.library.dto.OrderDetailsRequest.OrderDetailRequestDto;
+import com.academia.library.dto.OrderRequest;
+import com.academia.library.exception.BookNotFoundException;
+import com.academia.library.exception.OrderStatusException;
+import com.academia.library.model.Order;
+import com.academia.library.model.OrderStatus;
+import com.academia.library.repository.BookRepository;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.EnumUtils;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+@RequiredArgsConstructor
+public class OrderValidator {
+
+    private final BookRepository bookRepository;
+
+    public void validateBeforeCreate(OrderRequest orderRequest) {
+        String status = orderRequest.getStatus();
+        validateStringStatus(status);
+
+        validateBooks(orderRequest);
+    }
+
+    public void validateBeforeUpdate(OrderRequest orderRequest, Order order) {
+        validateBeforeUpdateStatus(order, OrderStatus.DRAFT);
+
+        validateBooks(orderRequest);
+    }
+
+    public void validateBeforeUpdateStatus(Order order, OrderStatus extendStatus){
+        OrderStatus orderStatus = order.getOrderStatus();
+
+        if (checkStatus(orderStatus, extendStatus)){
```

Comment:
I guess checkStatus method redundant here. please change to orderStatus != extendStatus

Reply:
Ok, thank